packages/frontend/src/components/markdown-container/chakra-ui-renderer.tsx

```diff
@@ -232,91 +274,53 @@ export const ChakraUIRenderer = (customRenderers: Renderers = {}, transformAsset
 
       return (
         <FetchInsightList
-          query={props.query}
+          query={query}
           sort={sort}
-          options={{ ...attributes, dispatchSearch: false }}
+          options={{ ...destringObject(props), dispatchSearch: false }}
           mb="1rem"
         />
       );
     },
-    vegaChart: ({ node, attributes }) => {
-      return <VegaRendererAsync specString={node.config} transformAssetUri={transformAssetUri} {...attributes} />;
-    },
-    video: ({ node, ...props }) => {
+    vegachart: ({ node, config, ...props }) => {
       return (
-        <VideoRenderer
-          url={props.url}
-          mimeType={getMimeForFileName(props.url)}
+        <VegaRendererAsync
+          key={hashCode(config)}
+          specString={config}
           transformAssetUri={transformAssetUri}
-          {...node.attributes}
+          {...props}
         />
       );
     },
-    xkcdChart: ({ node, attributes }) => {
-      return <XkcdChartRendererAsync type={node.xkcdType} configString={node.config} {...attributes} />;
-    },
-    footnoteReference: ({ node, ...props }) => {
+    video: ({ node, children, src, ...props }) => {
       return (
-        <Text as="sup" id={'reference-' + props.identifier}>
-          <ChakraLink href={'#footnote-' + props.identifier} fontWeight="500" color="frost.400">
-            [{props.label}]
-          </ChakraLink>
-        </Text>
-      );
-    },
-    footnoteDefinition: ({ node, children, ...props }) => {
-      return (
-        <Box pl="1.25rem" fontSize="sm">
-          <ChakraLink
-            id={'footnote-' + props.identifier}
-            href={'#reference-' + props.identifier}
-            fontWeight="500"
-            color="frost.400"
-          >
-            {props.label}
-          </ChakraLink>
-          :{' '}
-          <Box color="polar.500" display={children.length > 1 ? '' : 'inline-block'}>
-            {children}
-          </Box>
-        </Box>
-      );
-    },
-    customTable: ({ node, children }) => {
```

Comment:
Now the table directive converts directly into a `table` node and the box wrapper is provided there.

Reply:
Nice!